jwt/jwks_client.py

```diff
@@ -39,21 +64,32 @@ def get_signing_keys(self) -> List[PyJWK]:
 
     def get_signing_key(self, kid: str) -> PyJWK:
         signing_keys = self.get_signing_keys()
-        signing_key = None
-
-        for key in signing_keys:
-            if key.key_id == kid:
-                signing_key = key
-                break
+        signing_key = self.match_kid(signing_keys, kid)
 
         if not signing_key:
-            raise PyJWKClientError(
-                f'Unable to find a signing key that matches: "{kid}"'
-            )
+            # If no matching signing key from the cached jwk set, refresh the jwk set.
+            signing_keys = self.get_signing_keys(refresh=True)
```

Comment:
is it possible that the [first get_signing_keys()](https://github.com/wuhaoyujerry/pyjwt/pull/1/files#diff-0acaf974cb986e59e53c41333d4edacae0c55031720c2d2e066fc62dd0dfe7a5L41) already made a network call, as the cache expired, so this will make the same call again?

Reply:
Yeah, the call is the same, but it might return a different signing key set if the signing key sets just updated before the call.